packages/dd-trace/src/appsec/rasp/utils.js

```diff
@@ -30,13 +30,18 @@ class DatadogRaspAbortError extends Error {
 
 function handleResult (actions, req, res, abortController, config) {
   const generateStackTraceAction = actions?.generate_stack
-  if (generateStackTraceAction && config.appsec.stackTrace.enabled) {
+
+  const { enabled, maxDepth, maxStackTraces } = config.appsec.stackTrace
+
+  if (generateStackTraceAction && enabled) {
+    const frames = getCallsiteFrames(maxDepth)
```

Comment:
Before this PR, we weren't generating stack trace when we were over the max stack traces by request, with this change, we are generating stack trace (and adding its overhead) and sometimes, we we are over the limit, ignoring it.